number_theory/modular_unittest.cpp

```diff
@@ -181,5 +181,27 @@ TEST(ModularTest, IOStream) {
   test_modular_iostream<uint64_t>();
 }
 
+template <typename T>
+void test_modular_inverse() {
+  T num = 7;
+  T mod = 26;
+  EXPECT_EQ(inverse(num, mod), 15);
+
+  num = 8;
+  mod = 41;
+  EXPECT_EQ(inverse(num, mod), 36);
+}
+
+TEST(ModularTest, Inverse) {
+  test_modular_inverse<int8_t>();
+  test_modular_inverse<int16_t>();
+  test_modular_inverse<int32_t>();
+  test_modular_inverse<int64_t>();
+  // test_modular_inverse<uint8_t>();
```

Comment:
It seems the tests for unsigned types are commented because we can't pass an unsigned value to `exgcd`. The reason behind `exgcd` is that it can accept and return negative numbers. To pass an unsigned value to `exgcd`, you need to convert it to a signed value, and make sure it does not overflow. In this case, the modular inverse only makes sense when modular multiplication doesn't overflow, so it is safe to convert the value to a signed integer.

Reply:
I understand, but these signedness requirements forces users to cast their variables to signed/unsigned while calling functions from `tql` which get annoying.

Let's say that our function requires a variable to be signed/unsigned. Why don't we convert it to that type by ourselves and mention that the parameter must be signed/unsigned in documentation? That way the code will fail if they pass `int x = -1;` to a function that expects unsigned types since it will be converted to unsigned, but `int x = 12;` would still work as intended because the value stays the same after our conversion.